streams/test-utils/src/test/java/org/apache/kafka/streams/TopologyTestDriverTest.java

```diff
@@ -352,7 +373,9 @@ public void shouldCloseProcessor() {
         testDriver = new TopologyTestDriver(setupSingleProcessorTopology(), config);
 
         testDriver.close();
-        assertTrue(mockProcessors.get(0).closed);
+        if (!eosEnabled) {
```

Comment:
Why this? The `producer` must be close after the driver get's closed for both cases. (I think you added this, because `producer.close()` is not called atm when eos is enabled in `TestDriver#close()` as pointed out by my other comment. Thus, the failing test shows the bug and the fix is to close the producer instead of not checking if it was closed.

Reply:
Fixed.